boards/arm/nucleo_f334r8/doc/nucleof334r8.rst

```diff
@@ -0,0 +1,182 @@
+specialized shields. 
+
+The STM32 Nucleo board does not require any separate probe as it integrates the ST-LINK/V2-1
+debugger and programmer. 
+
+The STM32 Nucleo board comes with the STM32 comprehensive software HAL library together 
+with various packaged software examples.
+
+.. image:: img/nucleo_f334r8_board.jpg
+     :width: 500px
+     :height: 367px
+     :align: center
+     :alt: Nucleo F334R8
+
+More information about the board can be found at the `Nucleo F334R8 website`_.
+
+Hardware
+********
+Nucleo F334R8 provides the following hardware components:
+
+- STM32 microcontroller in QFP64 package
+- Two types of extension resources:
+    - Arduino™ Uno V3 connectivity
+    - ST morpho extension pin headers for full access to all STM32 I/Os
+- ARM® mbed™
+- On-board ST-LINK/V2-1 debugger/programmer with SWD connector:
+    - Selection-mode switch to use the kit as a standalone ST-LINK/V2-1
+- Flexible board power supply:
+    - USB VBUS or external source (3.3V, 5V, 7 - 12V)
+    - Power management access point
+- Three LEDs:
+    - USB communication (LD1), user LED (LD2), power LED (LD3)
+- Two push-buttons: USER and RESET
+- USB re-enumeration capability. Three different interfaces supported on USB:
+    - Virtual COM port
+    - Mass storage
+    - Debug port
+- Support of wide choice of Integrated Development Environments (IDEs) including:
+    - IAR™
+    - ARM® Keil®
+    - GCC-based IDEs
+
+More information about STM32F334R8 can be found here:
+       - `STM32F334 reference manual`_
+
+
+Supported Features
+==================
+
+The Zephyr nucleo_f334r8 board configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| CLOCK     | on-chip    | reset and clock control             |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | flash memory                        |
++-----------+------------+-------------------------------------+
+| IWDG      | on-chip    | independent watchdog                |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on Zephyr porting.
+
+The default configuration can be found in the defconfig file:
+
```

Comment:
ah, but is **that** much emphasis really needed for the filename?  I'll let you decide.